stories/div.stories.js

```diff
@@ -0,0 +1,136 @@
+//@flow
+import React from 'react';
+import styled from 'styled-components';
+import Div from '../src/lib/components/div/Div';
+import type { Props } from '../src/lib/components/div/Div';
+
+const control = {
+  control: {
+    type: 'select',
+    options: [0, 1, 2, 4, 8, 12, 14, 16, 20, 24, 28, 32],
+  },
+};
+
+export default {
+  title: 'Components/Div',
+  argTypes: {
+    m: control,
+    mt: control,
+    mr: control,
+    mb: control,
+    ml: control,
+    mx: control,
+    my: control,
+    p: control,
+    pt: control,
+    pr: control,
+    pb: control,
+    pl: control,
+    px: control,
+    py: control,
+  },
+};
+
+const HighlightBorder = styled.div`
+  border: 1px solid red;
+  div,
+  span,
+  p {
+    border: 1px solid blue;
+  }
+`;
+
+const GreenSpan = styled.span`
+  background-color: #0aada6;
+`;
+
+const BoldP = styled.p`
+  font-weight: 700;
+`;
+
+const DisplayTextComp = ({ text, ...rest }) => {
+  return (
+    <BoldP {...rest}>
+      <GreenSpan>{text}</GreenSpan>
+    </BoldP>
+  );
+};
+
+export const margin = () => {
+  return (
+    <HighlightBorder>
+      <Div m={0}>Props m 0</Div>
+      <Div m={1}>Props m 1</Div>
+      <Div m={2}>Props m 2</Div>
+      <Div m={4}>Props m 4 </Div>
+      <Div m={8}>Props m 8</Div>
+      <Div m={12}>Props m 12</Div>
+      <Div m={14}>Props m 14</Div>
+      <Div m={16}>Props m 16</Div>
+      <Div m={20}>Props m 20</Div>
+      <Div m={28}>Props m 28</Div>
+      <Div m={32}>Props m 32</Div>
+    </HighlightBorder>
+  );
+};
+
+export const padding = () => {
+  return (
+    <HighlightBorder>
+      <Div p={0}>Props p 0</Div>
+      <Div p={1}>Props p 1</Div>
+      <Div p={2}>Props p 2</Div>
+      <Div p={4}>Props p 4 </Div>
+      <Div p={8}>Props p 8</Div>
+      <Div p={12}>Props p 12</Div>
+      <Div p={14}>Props p 14</Div>
+      <Div p={16}>Props p 16</Div>
+      <Div p={20}>Props p 20</Div>
+      <Div p={28}>Props p 28</Div>
+      <Div p={32}>Props p 32</Div>
+    </HighlightBorder>
+  );
+};
+
+export const mix = () => {
+  return (
+    <HighlightBorder>
+      <Div pl={8} pt={12} mt={12}>
+        Props pl 8, pt 12, mt 12
+      </Div>
+      <Div mx={24} py={32}>
+        Props mx 24, py 32
+      </Div>
+      <Div ml={1} mt={2} mr={4} mb={8} pl={12} pt={14} pr={16} pb={24}>
+        Props ml 1, mt 2, mr 4, mb 8, pl 12, pt 14, pr 16, pb 24
+      </Div>
+    </HighlightBorder>
+  );
+};
+
+export const composition = () => {
+  return (
+    <HighlightBorder>
+      <Div as={GreenSpan} ml={16}>
+        Composition Example: Green background {'"</span>"'} with margin left
+      </Div>
+      <Div as={BoldP} my={14} p={14}>
+        Bold {'"</p>"'} with my 14 and p 14
+      </Div>
+      <Div
+        as={DisplayTextComp}
+        text={'Div + React component + nested styled component'}
+        m={12}
+        p={14}
+      ></Div>
```

Comment:
This is great, I'm wondering how does it affect the typing ? Do we get props autocompletion/validation for both Div and  DisplayTextComp component ?